library/src/main/java/com/pokegoapi/main/RequestHandler.java

```diff
@@ -313,4 +302,77 @@ public void run() {
 			}
 		}
 	}
+
+	private void setSignature(RequestEnvelope.Builder builder, ServerRequest[] serverRequests) {
+		byte[] uk22 = new byte[32];
+		new Random().nextBytes(uk22);
+
+		long curTime = api.currentTimeMillis();
+
+		byte[] authTicket = builder.getAuthTicket().toByteArray();
+
+		SignatureOuterClass.Signature.Builder sigBuilder = SignatureOuterClass.Signature.newBuilder()
+				.setLocationHash1(getLocationHash1(authTicket, builder))
+				.setLocationHash2(getLocationHash2(builder))
+				.setUnk22(ByteString.copyFrom(uk22))
+				.setTimestamp(api.currentTimeMillis())
+				.setTimestampSinceStart(curTime - api.startTime);
+
+
+		for (ServerRequest serverRequest : serverRequests) {
+			byte[] request = serverRequest.getRequest().toByteArray();
+			sigBuilder.addRequestHash(getRequestHash(authTicket, request));
+		}
+
+		// TODO: Call encrypt function on this
+		ByteString uk2 = sigBuilder.build().toByteString();
+		Unknown6OuterClass.Unknown6.newBuilder()
+				.setRequestType(6)
+				.setUnknown2(Unknown6OuterClass.Unknown6.Unknown2.newBuilder().setUnknown1(uk2));
+	}
+
+	private int getLocationHash1(byte[] authTicket, RequestEnvelope.Builder builder) {
```

Comment:
java uses signed int's. The python code unsigned.

Try to dump them both as hexadecimal (Integer.toHexString() in Java):

what is sent is exactly the same.